runtime/queries/python/highlights.scm

```diff
@@ -1,18 +1,27 @@
 ; Identifier naming conventions
 
+((identifier) @constant
+ (#match? @constant "^[A-Z_]*$"))
+
 ((identifier) @constructor
  (#match? @constructor "^[A-Z]"))
 
-((identifier) @constant
- (#match? @constant "^[A-Z][A-Z_]*$"))
+; Types
+
+((identifier) @type
+  (#match?
+    @type
+    "^(bool|bytes|dict|float|frozenset|int|list|set|str|tuple)$"))
+
+(type (identifier)) @type
 
 ; Builtin functions
 
 ((call
   function: (identifier) @function.builtin)
  (#match?
    @function.builtin
-   "^(abs|all|any|ascii|bin|bool|breakpoint|bytearray|bytes|callable|chr|classmethod|compile|complex|delattr|dict|dir|divmod|enumerate|eval|exec|filter|float|format|frozenset|getattr|globals|hasattr|hash|help|hex|id|input|int|isinstance|issubclass|iter|len|list|locals|map|max|memoryview|min|next|object|oct|open|ord|pow|print|property|range|repr|reversed|round|set|setattr|slice|sorted|staticmethod|str|sum|super|tuple|type|vars|zip|__import__)$"))
```

Comment:
These **are** built in functions though (if used with parentheses) https://www.programiz.com/python-programming/methods/built-in/str

VSCode highlights these differently, but I'd argue our highlighting is more precise here since we can distinguish between a type and a built-in function.

Reply:
That is true, however, I would respectfully argue that the type changing properties of these functions and their uses for type checking tools like Mypy and Typeguard more closely relate these particular functions to types.

At least in the codebases I am using, these functions are increasingly called to clarify types for type checkers as much as to change types at runtime.

Reply:
Independently of this (only just found this PR now), I implement some changes to the python highlighting. This touches on type highlighting among other things (parameters, decorators, etc.). I managed to treat "str" and the like as @function.builtin when being used as a function, and treat them as @type when being used in type hints. See #2457

Reply:
@Zeddicus414 if you think this is ready, go ahead and mark this PR as open again.

We've merged in the highlights work in #2457. archseer one of the changes keeps str, int, list, and so on as `@function.builtin` for the cases where they are being called, otherwise they are treated as `@type.builtin`. So hopefully that addressed your concern. Hopefully the sequence of commits is not too messy to follow, otherwise let us know if we need to clean things up.